core-bundle/src/DependencyInjection/Security/ContaoLoginFactory.php

```diff
@@ -69,4 +87,19 @@ protected function createAuthenticationFailureHandler($container, $id, $config):
     {
         return 'contao.security.authentication_failure_handler';
     }
+
+    private function createTwoFactorPreparationListener(ContainerBuilder $container, string $firewallName): void
+    {
+        $firewallConfigId = TwoFactorFactory::PROVIDER_PREPARATION_LISTENER_ID_PREFIX.$firewallName;
+
+        $container
+            ->setDefinition($firewallConfigId, new ChildDefinition(TwoFactorFactory::PROVIDER_PREPARATION_LISTENER_DEFINITION_ID))
+            ->replaceArgument(3, $firewallName)
+            ->replaceArgument(4, true)
+            ->replaceArgument(5, true)
+            ->addTag('kernel.event_listener', ['event' => 'security.authentication.success', 'method' => 'onLogin', 'priority' => PHP_INT_MAX])
+//            ->addTag('kernel.event_listener', ['event' => 'scheb_two_factor.authentication.require', 'method' => 'onAccessDenied'])
```

Comment:
Forgotten comment or TODO?

Reply:
Just some backup info that this is normally there, but the event is not fired in our case…

Reply:
added in f77abd61df4e9b54ae86efd8923e9ba082bd97de